### ISSUE TITLE
problem with downsample_layer.cpp/cu

### ISSUE THREAD

**Issue by PkuRainBow** [lmb-freiburg/flownet2#98]
Hi, I have compiled the program successfully..... However, I meet a problem when I want to train the Flying chairs dataset with FlowNet2-C, where I meet the following errors related the code of the Backpropagation.

line #132 - 139 / downsample_layer.cpp & downsample_layer.cu

```
 template <typename Dtype>
void DownsampleLayer<Dtype>::Backward_gpu(const vector<Blob<Dtype>*>& top,
      const vector<bool>& propagate_down, const vector<Blob<Dtype>*>& bottom) {

  for(int i=0; i<propagate_down.size(); i++) 
	if(propagate_down[i]) 
		LOG(FATAL) << "DownsamplingLayer cannot do backward.";
}
```

Here is the case of your provided prototxt.

I am wondering why no bp is allowed and how to solve this problem?

```
layer {
  name: "Downsample1"
  type: "Downsample"
  bottom: "scaled_flow_gt"
  bottom: "predict_flow6"
  top: "blob28"
}
```

**Comment by PkuRainBow** [comment]
@nikolausmayer

**Comment by nikolausmayer** [comment]
Hi,
the Downsample layer brings the groundtruth blob down to the same resolution as the predicted-flow blob. It does not do backpropagation because there is nothing to optimize here. The gradient information from the loss is backpropagated via the `predict_flow` bottom blob in the `flow_loss` layer, so it's ok to disable the pathway in the downsampling layer.

**Comment by nikolausmayer** [comment]
(closed due to inactivity)
